commit ebb94171cd1c13d9945967359415105edab6dc61 (refs/changes/30/1230/4)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2017-04-04T23:43:01+00:00 (2 years, 6 months ago)
    
    DAPI-330: Improved over-provisioning logic to take into account the currently used amounts
    DAPI-334: dapi can miscalculate disk and ram requirements when a disk or ram arg differs from pkg
    DAPI-335: Allow operator to specify a minimum free disk space threshold
    DAPI-338: DAPI hard-filter-invalid-servers algorithm dies on VMs in transient state
    DAPI-339: drop hard-filter-sick-servers algorithm
    Reviewed by: Angela Fong <angela.fong@joyent.com>

diff --git a/lib/designation.js b/lib/designation.js
index 8ad4977..3a94f8d 100644
--- a/lib/designation.js
+++ b/lib/designation.js
@@ -157,6 +157,7 @@ function getDefaults(changeDefaults) {
     setDefault('filter_vm_limit');
     setDefault('filter_docker_min_platform');
     setDefault('filter_owner_server');
+    setDefault('minimum_free_disk');
     setDefault('overprovision_ratio_cpu');
     setDefault('overprovision_ratio_ram');
     setDefault('overprovision_ratio_disk');
diff --git a/package.json b/package.json
index f4efff1..58be23f 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
     "assert-plus": "1.0.0",
     "async": "1.5.2",
     "bunyan": "1.8.5",
-    "dapi": "git+https://github.com/joyent/sdc-designation.git#2fbeb8d3ba939ab952c52f389c910a46f44b3094",
+    "dapi": "git+https://github.com/joyent/sdc-designation.git#f66ba1897f0eafcc3439d0af39555a3bdaebce4c",
     "deep-equal": "1.0.1",
     "dox": "0.4.1",
     "effluent-logger": "git+https://github.com/joshwilsdon/effluent-logger.git#d662f161a07f94045ad2afb45442931511c40e51",
diff --git a/sapi_manifests/cnapi/template b/sapi_manifests/cnapi/template
index 87c8d9a..283f667 100644
--- a/sapi_manifests/cnapi/template
+++ b/sapi_manifests/cnapi/template
@@ -63,6 +63,7 @@
 			"overprovision_ratio_cpu": "{{{ALLOC_OVERRIDE_OVERPROVISION_CPU}}}",
 			"overprovision_ratio_ram": "{{{ALLOC_OVERRIDE_OVERPROVISION_RAM}}}",
 			"overprovision_ratio_disk": "{{{ALLOC_OVERRIDE_OVERPROVISION_DISK}}}",
+			"minimum_free_disk": "{{{ALLOC_MIN_FREE_DISK}}}",
 			"weight_current_platform": "{{{ALLOC_WEIGHT_CURRENT_PLATFORM}}}",
 			"weight_next_reboot": "{{{ALLOC_WEIGHT_NEXT_REBOOT}}}",
 			"weight_num_owner_zones": "{{{ALLOC_WEIGHT_NUM_OWNER_ZONES}}}",
@@ -76,6 +77,7 @@
 			        "hard-filter-invalid-servers",
 			        "hard-filter-volumes-from",
 			        "hard-filter-reserved",
+			        "hard-filter-min-free-disk",
 			        "hard-filter-vlans",
 			        "hard-filter-platform-versions",
 			        "hard-filter-traits",
@@ -86,7 +88,6 @@
 			        "calculate-ticketed-vms",
 			        "hard-filter-capness",
 			        "hard-filter-vm-count",
-			        "hard-filter-sick-servers",
 			        "calculate-server-unreserved",
 			        "hard-filter-min-ram",
 			        "hard-filter-min-cpu",
